File: 2021/02/02.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func processCommand(pos *map[string]int, cmd string, amount int) {
	if cmd == "forward" {
		(*pos)["forward"] += amount
	} else if cmd == "down" {
		(*pos)["depth"] += amount
	} else if cmd == "up" {
		(*pos)["depth"] -= amount
	}
}

func main() {
	position := make(map[string]int)
	position["forward"] = 0
	position["depth"] = 0

	file, err := os.Open("input.txt")
	check(err)
	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		fields := strings.Fields(line)
		command := fields[0]
		amount, err := strconv.Atoi(fields[1])
		check(err)
		processCommand(&position, command, amount)
	}
	r := position["forward"] * position["depth"]
	fmt.Println("Result is", r)
}

File: 2021/02/02.py
position = {'forward': 0, 'depth': 0}

def process_line(line):
    direction, amount = line.split()
    amount = int(amount)
    if direction == "forward":
        position[direction] += amount
    elif direction == "up":
        position["depth"] -= amount
    elif direction == "down":
        position["depth"] += amount

with open("input.txt") as f:
    for line in f:
        process_line(line)

print(position['forward'] * position['depth'])
